### PR TITLE
[message] update comment/documentation

### PR DESCRIPTION
**Pull request by abtink**

### CODE REVIEW

**Comment by jwhui** [comment]
A [GitHub outage](https://status.github.com/messages) earlier this week caused inconsistencies that required a force-push to master.  Can you remove 0f2e87c from this PR?

File: src/core/common/message.hpp [hunk 1]
```diff
@@ -443,23 +443,25 @@ class Message : public Buffer
     int CopyTo(uint16_t aSourceOffset, uint16_t aDestinationOffset, uint16_t aLength, Message &aMessage) const;
 
     /**
-     * This method creates a copy of the current Message. It allocates the new one
-     * from the same Message Poll as the original Message and copies @p aLength octets of a payload.
+     * This method creates a copy of the current Message. It allocates the new one from the same message pool as the
```
Comment (jwhui) [inline src/core/common/message.hpp]: The "brief" should be on its own line.  More detailed description should be below.

File: src/core/common/message.hpp [hunk 2]
```diff
@@ -443,23 +443,25 @@ class Message : public Buffer
...
      *
      * @returns A pointer to the message or NULL if insufficient message buffers are available.
+     *
      */
     Message *Clone(uint16_t aLength) const;
 
     /**
-     * This method creates a copy of the current Message. It allocates the new one
-     * from the same Message Poll as the original Message and copies a full payload.
+     * This method creates a copy of the current Message. It allocates the new one from the same message pool as the
```
Comment (jwhui) [inline src/core/common/message.hpp]: The "brief" should be on its own line.  More detailed description should be below.

File: src/core/common/message.hpp [hunk 3]
```diff
@@ -443,23 +443,28 @@ class Message : public Buffer
     int CopyTo(uint16_t aSourceOffset, uint16_t aDestinationOffset, uint16_t aLength, Message &aMessage) const;
 
     /**
-     * This method creates a copy of the current Message. It allocates the new one
-     * from the same Message Poll as the original Message and copies @p aLength octets of a payload.
+     * This method creates a copy of the message.
      *
-     * The `Type`, `SubType`, `LinkSecurity` and `Priority` fields on the cloned message are also
+     * It allocates the new message from the same message pool as the original one and copies @p aLength octets
+     * of the payload. The `Type`, `SubType`, `LinkSecurity`, and `Priority` fields on the cloned message are also
```
Comment (jwhui) [inline src/core/common/message.hpp]: Should add `Offset` and `InterfaceId` to the list of cloned fields.

Comment (abtink) [reply src/core/common/message.hpp]: Somehow I totally missed this PR. Just noticed it today. Update the comment.

**Pull request merged**